### PR TITLE
little edits on review of the new sd jwt encoding 

### PR DESCRIPTION
**Pull request by bc-pi**
Some hopefully uncontroversial changes to the new approach from IETF 115 and a few other minor things I noticed.

### CODE REVIEW

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 1]
```diff
@@ -201,7 +201,7 @@ SD-JWT-DOC = (METADATA, SD-CLAIMS, NON-SD-CLAIMS)
 SD-JWT = SD-JWT-DOC | SIG(SD-JWT-DOC, ISSUER-PRIV-KEY)
 ```
 
-`SD-CLAIMS` is an object with claim names (`CLAIM-NAME`) mapped to the digests over the claim values (`CLAIM-VALUE`) with random salts (`SALT`). Digests are calculated using a digest derivation function such as a hash function, HMAC, or other (`DIGEST-DERIVATION()`):
+`SD-CLAIMS` is an array of digest values that and ensure the integrity of and map to the respective Disclosures.  Digest values are calculated over the Disclosures, each of which contains the claim name (`CLAIM-NAME`) and value (`CLAIM-VALUE`) with a random salt (`SALT`). Digests are calculated using a digest derivation function such as a hash function, HMAC, or other (`DIGEST-DERIVATION()`):
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:204]:
```suggestion
`SD-CLAIMS` is an array of digest values that ensures the integrity of and maps to the respective Disclosures.  Digest values are calculated over the Disclosures, each of which contains the claim name (`CLAIM-NAME`), the claim value (`CLAIM-VALUE`), and a random salt (`SALT`). Digests are calculated using a digest derivation function such as a hash function, HMAC, or other (`DIGEST-DERIVATION()`):
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 2]
```diff
@@ -214,8 +214,6 @@ However, the term "salt" is used throughout this document for brevity.
 
 `SD-CLAIMS` can also be nested deeper to capture more complex objects, as will be shown later.
 
-`SD-JWT` is sent from the Issuer to the Holder, together with the mapping of the plain-text claim values, the salt values, and potentially some other information.
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:217]: Was this perceived redundant?

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 3]
```diff
@@ -758,7 +756,7 @@ checks the Disclosures correctly.
 
 ## Entropy of the salt {#salt-entropy}
 
-The security model relies on the fact that the salt is not learned or guessed by
+The security model that conceals the cleartext claims relies on the fact that the salt cannot be learned or guessed by
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:759]:
```suggestion
The security model that conceals the plaintext claims relies on the fact that the salt cannot be learned or guessed by
```

**Review by Sakurann — APPROVED** [review]

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 1 above)
Comment (danielfett) [reply draft-ietf-oauth-selective-disclosure-jwt.md:204]: Best of both?
```suggestion
`SD-CLAIMS` is an array of digest values that ensure the integrity of and map to the respective Disclosures.  Digest values are calculated over the Disclosures, each of which contains the claim name (`CLAIM-NAME`), the claim value (`CLAIM-VALUE`), and a random salt (`SALT`). Digests are calculated using a digest derivation function such as a hash function, HMAC, or other (`DIGEST-DERIVATION()`):
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 4]
```diff
@@ -461,12 +459,12 @@ Claims that are not selectively disclosable are included in the SD-JWT in plaint
...
 Selectively disclosable claims are omitted from the SD-JWT. Instead, the hash digests of the respective Disclosures and potentially decoy digests are contained as an array in a new JWT claim, `_sd`.
 
-The `_sd` claim MUST be an array of strings, each string being a hash digests of a Disclosure or a decoy digest as described above.
+The `_sd` claim MUST be an array of strings, each string being a hash digest of a Disclosure or a decoy digest as described above.
 The array MAY be empty in case the Issuer decided not to selectively disclose any of the claims at that level. However, it is RECOMMENDED to omit `_sd` claim in this case to save space.
 
 The Issuer MUST hide the original order of the claims in the array. To ensure this, it is RECOMMENDED to shuffle the array of hashes, e.g., by sorting it alphanumerically or randomly. The precise method does not matter as long as it does not depend on the original order of elements.
 
-Issuers MUST NOT issue SD-JWTs where
+Issuers MUST NOT issue SD-JWTs where:
```
Comment (danielfett) [inline draft-ietf-oauth-selective-disclosure-jwt.md:467]: I did not put a colon here as the introduction to the list is not a complete sentence.

```suggestion
Issuers MUST NOT issue SD-JWTs where
```

**Review by danielfett — APPROVED** [review]
Thank you, a number of good points here! Generally approved, left a few comments.

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 3 above)
Comment (bc-pi) [reply draft-ietf-oauth-selective-disclosure-jwt.md:759]: The draft currently uses plaintext, cleartext and plain text. Maybe a full reconciliation is needed at some point. In the meantime, I'll take this suggestion.

File: draft-ietf-oauth-selective-disclosure-jwt.md (hunk 2 above)
Comment (bc-pi) [reply draft-ietf-oauth-selective-disclosure-jwt.md:217]: Yeah, it felt redundant and somewhat out of order with the content just below.

**Pull request merged**